calico/common.py

```diff
@@ -71,18 +75,27 @@
     "protocol",
     "src_net",
     "src_tag",
+    "src_selector",
     "src_ports",
     "dst_net",
     "dst_tag",
+    "dst_selector",
     "dst_ports",
     "icmp_type",
     "icmp_code",
     "ip_version",
 ])
 
+# Valid actions to see in a rule.
+KNOWN_ACTIONS = set(["allow", "deny", "next-tier"])
+
 # Regex that matches only names with valid characters in them. The list of
 # valid characters is the same for endpoints, profiles, and tags.
-VALID_ID_RE = re.compile('^[a-zA-Z0-9_\.\-]+$')
+VALID_ID_RE = re.compile(r'^[a-zA-Z0-9_\.\-]+$')
+
+# Regex for validating the names of labels, which need to be rich enough to
+# allow for Kubernetes implementation, for example.
+VALID_LABEL_NAME_RE = re.compile(r'^[%s]+$' % re.escape(LABEL_CHARS))
```

Comment:
Good spot; I checked the docs and I believe this is sound.  I also checked using the REPL and I get
```
>>> re.escape("_.-/")
'\\_\\.\\-\\/'

>>> re.match("[%s]" % re.escape("_.-/"), '-')
<_sre.SRE_Match object at 0x7f9d51c83920>
>>> re.match("[%s]" % re.escape("_.-/"), '/')
<_sre.SRE_Match object at 0x7f9d51c69988>
>>> re.match("[%s]" % re.escape("_.-/"), '.')
<_sre.SRE_Match object at 0x7f9d51c2f920>
>>> re.match("[%s]" % re.escape("_.-/"), '_')
<_sre.SRE_Match object at 0x7f9d51c83920>
>>> re.match("[%s]" % re.escape("_.-/"), '\\')
```

I.e. we end up with a `\` before each special character but the regex still doesn't match `\` itself (but it does match the characters that were escaped).

The docs are explicit that `[a\-z]` matches only `a`, `z` and `-`; i.e. you can escape a `-` between `[]`s